apps/televisit-demo/frontend/src/backend/serverless/template.yaml

```diff
@@ -0,0 +1,385 @@
+    Properties:
+      CompatibleRuntimes:
+        - "nodejs12.x"
+      Content:
+        S3Bucket: amazon-chime-blog-assets
+        S3Key: AWS_SDK_CHIME_LAYER.zip
+
+  #Lambda that creates AWS Chime App instance
+  ChimeAppInstanceLambda:
+      Type: "AWS::Lambda::Function"
+      Properties:
+        Handler: "index.handler"
+        Role: !GetAtt LambdaExecuteRole.Arn
+        Runtime: "nodejs12.x"
+        Timeout: 60
+        Layers: 
+          - !Ref AWSSDKChimeLayer
+        Code:
+            ZipFile: >
+                "use strict";
+                const AWS = require("aws-sdk");
+                const uuidv4 = require("uuid");
+                var response = require("cfn-response"); 
+                AWS.config.update({ region: process.env.AWS_REGION });
+                const chime = new AWS.Chime({ region: process.env.AWS_REGION }); 
+                
+                exports.handler = async (event, context, callback) => {
+                  console.log("Event: \n", event);
+                  console.log("Create Chime SDK App Instance");
+                  if (event["RequestType"] === "Create") {
+                    //create a chime app instance
+                    var params = {
+                      Name: `AWSChimeMessagingSDKDemo-${uuidv4()}`,
+                    };
+                    try {
+                      var chime_response = await chime.createAppInstance(
+                        params,
+                        function (err, data) {
+                          if (err) console.log(err, err.stack);
+                          // an error occurred
+                          else {
+                            console.log(data); // successful response
+                            return data;
+                          }
+                        }
+                      ).promise();;
+                      await response.send(event, context, response.SUCCESS, chime_response);
+                    } catch (error) {
+                      console.log("ERROR CAUGHT \n", error);
+                      await response.send(event, context, response.FAILED, {});
+                    }
+                  } else {
+                    //NOT A CREATE REQUEST
+                    await response.send(event, context, response.SUCCESS, {});
+                  }
+                };
+
+  # Trigger Lambda function to create Amazon Chime App Instance creation
+  TriggerChimeAppInstanceLambda:
+      Type: AWS::CloudFormation::CustomResource
+      Properties: 
+        ServiceToken: !GetAtt ChimeAppInstanceLambda.Arn
+
+  # Creates an S3 bucket to store chat attachments
+  ChatAttachmentsBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      CorsConfiguration:
+        CorsRules:
+          - AllowedHeaders:
+              - "*"
+            AllowedMethods:
+              - GET
+              - HEAD
+              - PUT
+              - POST
+              - DELETE
+            AllowedOrigins:
+              - "*"

```

Comment:
non-blocking:
can this be a param in the future?

Reply:
yes